File: interviews/leetcode/334. Increasing Triplet Subsequence/solution.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        first = float('inf')  # Initialize the first smallest element
        second = float('inf') # Initialize the second smallest element

        for num in nums:
            if num <= first:       # Update first if num is smaller
                first = num
            elif num <= second:    # Update second if num is larger than first but smaller than second
                second = num
            else:                  # If we find a num larger than both, we have a triplet
                return True
        return False

File: interviews/leetcode/334. Increasing Triplet Subsequence/solution.go

func increasingTriplet(nums []int) bool {
	first, second := int(^uint(0)>>1), int(^uint(0)>>1) // Initialize to maximum integer value

	for _, num := range nums {
		if num <= first { // Update 'first' if num is smaller
			first = num
		} else if num <= second { // Update 'second' if num is larger than 'first'
			second = num
		} else { // Found a number larger than both
			return true
		}
	}
	return false
}

/*
func main() {
    // Test cases
    fmt.Println(increasingTriplet([]int{1, 2, 3, 4, 5})) // Output: true
    fmt.Println(increasingTriplet([]int{5, 4, 3, 2, 1})) // Output: false
    fmt.Println(increasingTriplet([]int{2, 1, 5, 0, 4, 6})) // Output: true
}
*/
